src/ast/ast_validations.c

```
@@ -1,2478 +1,116 @@
-		   root_type == CYPHER_AST_COMMENT) {
-			continue;
-		} else if(root_type != CYPHER_AST_STATEMENT) {
-			ErrorCtx_SetError(EMSG_UNSUPPORTED_QUERY_TYPE, cypher_astnode_typestr(root_type));
-			return AST_INVALID;
-		} else {
-			// We got a statement.
-			*index = i;
-			return AST_VALID;
+// Check if two label sets conflict (are non-identical)
+static int _LabelsConflict(array labels1, array labels2) {
+	uint len1 = array_len(labels1);
```

Cppcheck (2.21.0)

[error] Common realloc mistake

(memleakOnRealloc)